### PR TITLE
[Issue 10996]  release multiTopicsConsumerImpl when unsubscribe

### PR DESCRIPTION
**Pull request by Shoothzj**
Fixes 10996

### Motivation
MutliConsumer not released after calling method unsubscribe, and can't be released by other method.

### Modifications

Make MutliConsumer release completely on method unsuscribe.

### Verifying this change

Same test as in ISSUE 10996, and jmap shows nothing.

image

### CODE REVIEW

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImpl.java [hunk 1]
```diff
@@ -560,12 +560,9 @@ public void negativeAcknowledge(MessageId messageId) {
             .whenComplete((r, ex) -> {
                 if (ex == null) {
                     setState(State.Closed);
-                    unAckedMessageTracker.close();
+                    cleanupMultiConsumer();
                     closeFuture.complete(null);
                     log.info("[{}] [{}] Closed Topics Consumer", topic, subscription);
```
Comment (315157973) [inline pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImpl.java:565]: This log seems to be repeated

**Comment by Anonymitaet** [comment]
@Shoothzj thanks for your contribution. For this PR, do we need to update docs?

**Comment by Shoothzj** [comment]
> @Shoothzj thanks for your contribution. For this PR, do we need to update docs?

We don't need to update docs. Cause user can't not call `unsubscribe` and `close` at the same time.  People who call `unsubscribe` will out of memory.

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImpl.java (hunk 1 above)
Comment (Shoothzj) [reply pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImpl.java]: Your are right. Fixed

**Comment by gaoran10** [comment]
Only need to change docs?

**Comment by Shoothzj** [comment]
> Only need to change docs?

Sorry to push the wrong commit.

**Review by merlimat — APPROVED** [review]

**Pull request merged**